[Docs] Update demo to be a meta project spec!

seshrs opened the pull request:
The current demo page at https://eecs485staff.github.io/primer-spec does not really demonstrate the power of Primer Spec, especially because it's a short page and a contrived example.

This PR changes the demo page into a completely new project spec. Not only that, it's meta — the project guides students towards building Primer Spec, complete with a starter file that includes the entire spec itself!

Although the style of the spec is influenced by the other EECS 485 specs, it's not _truly_ a project spec, rather more of a guide and an advertisement of Primer Spec's features. I guess that's the point of a demo page anyway 🤷‍♂.

Also added a "Users" section to the README with a list of courses that currently use Primer Spec. Hopefully, next semester might feature two more courses: EECS 482 and EECS 370!

Suggestions for wording and content would be appreciated. 😄 

Closes #14.
Update: Also resolves #29 with changes from 2869aea16ae9d8c1d00436af018b90dc63df09c2.

awdeorio commented:
Great idea!  I'll do my best to do a detailed review soon.

seshrs commented:
Thanks! And please take your time.

awdeorio commented on README.md:
```diff
@@ -10,6 +10,7 @@ Primer Spec is built on top of the wonderful [Primer theme](https://github.com/p
 - [Usage](#usage)
   - [Using with Jekyll](#using-with-jekyll)
```
Something that's confusing about the README are the terms "Jekyll" and "Markdown".  I think you're using them interchangeably.  Perhaps "Using with Markdown and Jekyll"?  This confusion also caught me when reading the opening description of the README.

seshrs replied:
Hmm, I see how this could be confusing. This project is technically a Jekyll theme (since it can be rendered with the Jekyll Remote Theme plugin), but will also be distributed as a standalone JavaScript plugin that formats any compatible HTML page.

Adding to the confusion: Jekyll has a built-in Markdown-to-HTML converter, but it's not necessary to use it. All of our project specs are written in Markdown for easy maintenance.

Maybe it would be cleaner to simply describe Primer Spec as a Jekyll theme with support for Markdown? (Especially since we probably want to recommend the Jekyll+Markdown approach to new courses.) Then, deeper in the README, we could describe how it could also be used without Jekyll.

seshrs replied:
I think you're right — we should probably always recommend new users to start with Jekyll (and Markdown). The HTML/JS plugin covers a slightly more niche use-case.

awdeorio commented on README.md:
```diff
@@ -10,6 +10,7 @@ Primer Spec is built on top of the wonderful [Primer theme](https://github.com/p
 - [Usage](#usage)
   - [Using with Jekyll](#using-with-jekyll)
   - [Using the plugin](#using-the-plugin)
```
Perhaps "Using the HTML JavaScript plugin"?

awdeorio commented on index.md:
```diff
@@ -2,131 +2,201 @@
 layout: spec
```
Maybe I'm missing something, but I don't think this is linked from the main page AKA `README.md`.  Perhaps a good way to structure the linking would be to structure the README like this:

- Using
    - HTML/JavaScript
    - Markdown/Jekyll
    - Walk through example

The walk through example could link to the meta project that you added.

seshrs replied:
Makes sense! Yes, I missed to include this demo in the README.

seshrs replied:
Actually I had included it at the top of the README. I've also referenced it at the end of the "Jekyll + MarkDown" section. (I wasn't sure if I had enough content to create a "walk through example" section.

awdeorio commented on index.md:
```diff
@@ -2,131 +2,201 @@
 
-* * *
+#### Where do I put the sidebar code?
+We added our sidebar HTML at the very top of the `body` tag, just before the the main content begins. (The main content is contained in a `div` with class `markdown-body`.)
```
Is the idea that I will write primer-spec from scratch?  Or should I copy-paste code from the repo at this point?

seshrs replied:
I personally have been viewing the project as more of a gimmick, and don't really expect anyone to actually work on it (unless some dedicated 485 students discover it 😆).

When writing this part, I expected a dedicated reader to implement this on their own from scratch, while drawing ideas for implementation from the current behavior of Primer Spec.

seshrs commented:
@awdeorio: Based on your feedback, I decided that we should always recommend new users to start using Primer Spec with GitHub Pages + Jekyll. I moved all other instructions and documentation to `docs/USAGE_ADVANCED.md`. This change also simplified the README.

What do you think of the new changes to the README?

awdeorio commented on README.md:
````diff
@@ -2,130 +2,52 @@
     ```
 
-6. If you're using GitHub pages, follow the steps in the [USAGE_JEKYLL file](docs/USAGE_JEKYLL.md#Previewing-GitHub-Pages-locally) to preview your website locally.
+4. Deploy your site with GitHub pages!
````
Maybe link to the preview locally here?

seshrs replied:
I added a link to the preview right below this line. I've updated the wording in 35dabbfd5fdbbf94a494b88860ba0fe3276fde50 to hopefully be more explicit.

seshrs commented:
I'm going to merge this in for now, but let me know if you have further suggestions!
